crypto/bls12381/g1.go

```diff
@@ -0,0 +1,436 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package bls12381
+
+import (
+	"errors"
+	"math"
+	"math/big"
+)
+
+// PointG1 is type for point in G1.
+// PointG1 is both used for Affine and Jacobian point representation.
+// If z is equal to one the point is considered as in affine form.
+type PointG1 [3]fe
+
+func (p *PointG1) Set(p2 *PointG1) *PointG1 {
+	p[0].set(&p2[0])
+	p[1].set(&p2[1])
+	p[2].set(&p2[2])
+	return p
+}
+
+// Zero returns G1 point in point at infinity representation
+func (p *PointG1) Zero() *PointG1 {
+	p[0].zero()
+	p[1].one()
+	p[2].zero()
+	return p
+}
+
+type tempG1 struct {
+	t [9]*fe
+}
+
+// G1 is struct for G1 group.
+type G1 struct {
+	tempG1
+}
+
+// NewG1 constructs a new G1 instance.
+func NewG1() *G1 {
+	Init()
+	t := newTempG1()
+	return &G1{t}
+}
+
+func newTempG1() tempG1 {
+	t := [9]*fe{}
+	for i := 0; i < 9; i++ {
+		t[i] = &fe{}
+	}
+	return tempG1{t}
+}
+
+// Q returns group order in big.Int.
+func (g *G1) Q() *big.Int {
+	return new(big.Int).Set(q)
+}
+
+func (g *G1) fromBytesUnchecked(in []byte) (*PointG1, error) {
+	p0, err := fromBytes(in[:48])
+	if err != nil {
+		return nil, err
+	}
+	p1, err := fromBytes(in[48:])
+	if err != nil {
+		return nil, err
+	}
+	p2 := new(fe).one()
+	return &PointG1{*p0, *p1, *p2}, nil
+}
+
+// FromBytes constructs a new point given uncompressed byte input.
+// FromBytes does not take zcash flags into account.
+// Byte input expected to be larger than 96 bytes.
+// First 96 bytes should be concatenation of x and y values.
+// Point (0, 0) is considered as infinity.
+func (g *G1) FromBytes(in []byte) (*PointG1, error) {
+	if len(in) < 96 {
```

Comment:
Here you only check that `len(in)` is larger than or equal to 96. In the method doc you mention that it should be larger (not equal). But below on line 100 `p1, err := fromBytes(in[48:])` will fail on any input larger than 96 because fromBytes explicitly required exactly 48 byte.

Doc says > 96
Check says >= 96
Implementation says == 96

Please fix